src/darray.jl

```diff
@@ -784,3 +784,12 @@ function Base.fill!(A::DArray, x)
     end
     return A
 end
+
+using Random
+
+function Random.rand!(rng::AbstractRNG, A::DArray, ::Type{T}) where T
+    asyncmap(procs(A)) do p
+        remotecall_wait((rng, A, T)->rand!(rng, localpart(A), T), p, rng, A, T)
```

Comment:
Won't this give the exact same variates on each worker since the same `RNG` is sent to each of them?

Reply:
Hm yes... Should we for now just limit this to `rand!(DArray, T)`?

Reply:
I think that would be better. Explicitly handling `RNG` objects in a distributed setting seems like an open problem for us.